src/Simplify.elm

```diff
@@ -4512,6 +4523,21 @@ listDropChecks checkInfo =
                 []
 
 
+listMapNChecks : { n : Int } -> CheckInfo -> List (Error {})
+listMapNChecks { n } checkInfo =
+    if List.any (\(Node _ expression) -> expression == Expression.ListExpr []) checkInfo.argsAfterFirst then
+        [ Rule.errorWithFix
+            { message = "Using List.map" ++ String.fromInt n ++ " with any list being [] will result in []"
+            , details = [ "You can replace this call by []." ]
+            }
+            checkInfo.fnRange
+            [ Fix.replaceRangeBy checkInfo.parentRange "[]" ]
```

Comment:
The fix is incorrect for when there are less arguments than necessary. For instance
```elm
List.map2 f []
```
currently gets fixed to
```elm
[]
```
but should be something like
```elm
(\_ -> [])
```
with one `_` per missing argument. Or similarly using `always`.

I think using `always []` would be a fine fix if there is one missing argument, but I'm unsure what would be "nice enough" when there are more missing parameters. I'm thinking having `always (always (always []))` would be pretty unreadable, so in that case `\_ _ _ -> []` would be nicer? Or we could avoid writing a fix?

Let's add a unit test for this as well.

Reply:
Yes. Also good catch with the parens which are required if the `List.mapN` call is right-pizza-ed into another function

Reply:
I'm not fond of `always` myself, but mostly because I would rarely write it myself.